pkg/utils/experimentHelpers.go

```diff
@@ -109,31 +120,37 @@ func (expDetails *ExperimentDetails) SetEnvFromEngine(engineName string, clients
 }
 
 // SetLabels sets the Experiment Labels, in Experiment Structure
-func (expDetails *ExperimentDetails) SetLabels(clients ClientSets) {
+func (expDetails *ExperimentDetails) SetLabels(engine *EngineDetails, clients ClientSets) error {
 	expirementSpec, err := clients.LitmusClient.LitmuschaosV1alpha1().ChaosExperiments(expDetails.Namespace).Get(expDetails.Name, metav1.GetOptions{})
 	if err != nil {
 		klog.V(0).Infoln(err)
+		return err
 	}
 	expDetails.ExpLabels = expirementSpec.Spec.Definition.Labels
-
+	expDetails.ExpLabels["chaosUID"] = engine.UID
+	return nil
 }
 
 // SetImage sets the Experiment Image, in Experiment Structure
-func (expDetails *ExperimentDetails) SetImage(clients ClientSets) {
+func (expDetails *ExperimentDetails) SetImage(clients ClientSets) error {
 	expirementSpec, err := clients.LitmusClient.LitmuschaosV1alpha1().ChaosExperiments(expDetails.Namespace).Get(expDetails.Name, metav1.GetOptions{})
 	if err != nil {
 		klog.V(0).Infoln(err)
+		return err
 	}
 	expDetails.ExpImage = expirementSpec.Spec.Definition.Image
+	return nil
 }
 
 // SetArgs sets the Experiment Image, in Experiment Structure
-func (expDetails *ExperimentDetails) SetArgs(clients ClientSets) {
+func (expDetails *ExperimentDetails) SetArgs(clients ClientSets) error {
 	expirementSpec, err := clients.LitmusClient.LitmuschaosV1alpha1().ChaosExperiments(expDetails.Namespace).Get(expDetails.Name, metav1.GetOptions{})
 	if err != nil {
 		klog.V(0).Infoln(err)
```

Comment:
Please remove klog instances, and use errors.Wrapf to wrap the errors, and then return it as the error